doc/source/administrator/authentication.md

```diff
@@ -0,0 +1,427 @@
+sharing, publishing and discovering data, whether your files live on a
+supercomputer, lab cluster, tape archive, public cloud, or your own
+laptop. Start a Globus app [here](https://developers.globus.org/)!
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    GlobusOAuthenticator:
+      identity_provider: your-university.edu
+    JupyterHub:
+      authenticator_class: oauthenticator.globus.GlobusOAuthenticator
+```
+
+#### Azure Active Directory
+
+Azure Active Directory
+\<<https://docs.microsoft.com/en-us/azure/active-directory/>>`_ is an
+identity provider from Microsoft Azure. The main additional option to
+configure for Azure AD from any other oauth provider is the tenant id.
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-aad-client-id
+      client_secret: your-aad-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    AzureAdOAuthenticator:
+      tenant_id: your-aad-tenant-id
+    JupyterHub:
+      authenticator_class: oauthenticator.azuread.AzureAdOAuthenticator
+```
+
+#### OpenID Connect
+
+[OpenID Connect](https://openid.net/connect) is an identity layer on top
+of the OAuth 2.0 protocol, implemented by [various servers and
+services](https://openid.net/developers/certified/#OPServices). While
+OpenID Connect endpoint discovery is not supported by oauthentiator, you
+can still configure JupyterHub to authenticate with OpenID Connect
+providers by specifying all endpoints in GenericOAuthenticator. By
+setting `login_service` you can customize the label on the
+login button.
+
+Here's an example for authenticating against
+[keycloak](https://www.keycloak.org/docs/latest/securing_apps/index.html#endpoints),
+after you [configure an OIDC
+Client](https://www.keycloak.org/docs/latest/server_admin/index.html#oidc-clients)
+and obtain the confidential client credentials.
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+      authorize_url: https://${host}/auth/realms/${realm}/protocol/openid-connect/auth
+      token_url: https://${host}/auth/realms/${realm}/protocol/openid-connect/token
+      userdata_url: https://${host}/auth/realms/${realm}/protocol/openid-connect/userinfo
+    GenericOAuthenticator:
+      login_service: keycloak
+      username_key: preferred_username
+      userdata_params:
+        state: state
+    JupyterHub:
+      authenticator_class: oauthenticator.generic.GenericOAuthenticator
+```
+
+#### Auth0
+
+Auth0 is a popular commercial provider of identity management. The
+JupyterHub helm chart does not include support for Auth0 by default. To
+use Auth0, `extraEnv` and `extraConfig` must be configured as follows:
```

Comment:
You've mentioned `extraEnv` and `extraConfig` but there's no sign of it below, maybe leftover from a copy and paste?

Reply:
Ah, I removed extraEnv and perhaps also extraConfig without updating the text. I'll update the text to match.